vsintegration/src/FSharp.Editor/CodeLens/FSharpCodeLensService.fs

```diff
@@ -186,22 +184,30 @@ type internal FSharpCodeLensService
                                 logWarningf "Couldn't acquire CodeLens data for function %A" func
                                 return None
                         else return None
+#if DEBUG
                     with e -> 
                         logErrorf "Error in lazy line lens computation. %A" e
+#else
+                    with _ ->
+#endif
                         return None
                 }
             
             let inline setNewResultsAndWarnIfOverriden fullDeclarationText value = 
                 if newResults.ContainsKey fullDeclarationText then
+#if DEBUG
                     logWarningf "New results already contains: %A" fullDeclarationText
+#else
+                    ()
+#endif
                 newResults.[fullDeclarationText] <- value
-
             for symbolUse in symbolUses do
                 if symbolUse.IsFromDefinition then
                     match symbolUse.Symbol with
                     | :? FSharpMemberOrFunctionOrValue as func when func.IsModuleValueOrMember || func.IsProperty ->
-                        let funcID = func.FullName
-                        let fullDeclarationText = funcID
+                        let funcID = func.LogicalName + (func.FullType.ToString() |> hash |> string)
```

Comment:
Why was `func.FullName` not sufficient?

Reply:
func.FullName is ambigious if only the types change and not the function name.

The new algorithm for distributing unqiue ids is using the logical name and the type name (as hash to short it).